akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -241,6 +241,9 @@ final class Flow[-In, +Out, +Mat](private[stream] override val module: Module)
 
   /** Converts this Scala DSL element to it's Java DSL counterpart. */
   def asJava: javadsl.Flow[In, Out, Mat] = new javadsl.Flow(this)
+
+  override def toString =
+    s"""|Flow(${module.toString})""".stripMargin
```

Comment:
What do you mean ugly? It can get pretty long, but if someone is printing out flows, then it probably means she is debugging. In that case verbose output is helpful. Here is the output of one flow created from graph, and another from ops:
```
Flow(
 Module: nested
 Modules: 
   akka.stream.impl.StreamLayout$CompositeModule
 Downstreams: 
   Flow.out -> Merge.in0
   Merge.out -> Broadast.in
   Broadcast.out0 -> Flow.in
   Flow.out -> SubscriberSink.in
 Upstreams: 
   Merge.in0 -> Flow.out
   Broadast.in -> Merge.out
   Flow.in -> Broadcast.out0
   SubscriberSink.in -> Flow.out
)
Flow(
 Module: filter
 Modules: 
   filter
   map
   take
 Downstreams: 
   Flow.out -> Flow.in
   Flow.out -> Flow.in
 Upstreams: 
   Flow.in -> Flow.out
   Flow.in -> Flow.out
)
```